### PR TITLE
Partial code coverage fix

### LINKED ISSUE #1: PrimeHashingFamily UnitTests
After running the fine code coverage, we found out that the class haven't been tested yet.

### PR DESCRIPTION
**Pull request by gyupetosz**
100% code coverage achieved on PrimeHashing family class.

### CODE REVIEW

File: UnitTest/DataStructuresTests/HashTableSeparateChainingTest.cs [hunk 1]
```diff
@@ -147,4 +147,4 @@ public static void CopyTo_EmptyHashTable_ReturnsSuccessful()
             studentsMarks.CopyTo(array, 0);
         }
     }
-}
+}
```
Comment (AkosUzonyi) [inline UnitTest/DataStructuresTests/HashTableSeparateChainingTest.cs]: Please remove every change not related to the new unit test

File: UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+namespace UnitTest.DataStructuresTests
+{
+    
+    public static class PrimeHashingFamilyTest
+    {
+        [Fact]
+        public static void CheckHashIdenticalInt()
+        {
+            PrimeHashingFamily hash1 = new PrimeHashingFamily(1);
+            PrimeHashingFamily hash2 = new PrimeHashingFamily(1);
```
Comment (AkosUzonyi) [inline UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs]: We could test with a bit larger number of hash functions than 1.

File: UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            int hashValue = "főzelék".GetHashCode();
+
+            for (int i = 1; i < hash1.NumberOfFunctions; i++)
+            {
+                hash1 = new PrimeHashingFamily(i);
+                hash2 = new PrimeHashingFamily(i);
```
Comment (AkosUzonyi) [inline UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs]: These assignments shouldn't be here, hash1 and hash2 are already initialized.

File: UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+{
+    
+    public static class PrimeHashingFamilyTest
+    {
+        [Fact]
+        public static void CheckHashIdenticalInt()
+        {
+            PrimeHashingFamily hash1 = new PrimeHashingFamily(1);
+            PrimeHashingFamily hash2 = new PrimeHashingFamily(1);
+            int hashValue = "főzelék".GetHashCode();
```
Comment (AkosUzonyi) [inline UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs:17]: Hami :D

File: UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        public static void CheckHashIdenticalString()
+        {
+            PrimeHashingFamily hash1 = new PrimeHashingFamily(10);
+            PrimeHashingFamily hash2 = new PrimeHashingFamily(10);
+            string hashValue = "főzelék";
+
+            for (int i = 1; i < hash1.NumberOfFunctions; i++)
+            {
+                hash1 = new PrimeHashingFamily(i);
+                hash2 = new PrimeHashingFamily(i);
```
Comment (AkosUzonyi) [inline UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs]: Likewise

**Review by AkosUzonyi — CHANGES_REQUESTED** [review]

File: UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        [Fact]
+        public static void CheckHashIdenticalInt()
+        {
+            PrimeHashingFamily hash1 = new PrimeHashingFamily(1);
+            PrimeHashingFamily hash2 = new PrimeHashingFamily(1);
+            int hashValue = "főzelék".GetHashCode();
+
+            for (int i = 1; i < hash1.NumberOfFunctions; i++)
+            {
+                hash1 = new PrimeHashingFamily(i);
```
Comment (MarcellHarmaci) [inline UnitTest/DataStructuresTests/PrimeHashingFamilyTest.cs]: This is changing the condition of the for loop inside the loop, which is a bad practice.

**Review by MarcellHarmaci — CHANGES_REQUESTED** [review]

**Review by AkosUzonyi — APPROVED** [review]

**Review by AkosUzonyi — CHANGES_REQUESTED** [review]
Please add a short documentation of this change to `doc/unit_test.md`

**Review by MarcellHarmaci — APPROVED** [review]

**Pull request merged**